pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -292,7 +294,7 @@ ContextImpl setupContext() throws PulsarClientException {
             Thread.currentThread().setContextClassLoader(functionClassLoader);
             return new ContextImpl(instanceConfig, instanceLog, client, secretsProvider,
                 collectorRegistry, metricsLabels, this.componentType, this.stats, stateManager,
-                pulsarAdmin, clientBuilder, fatalHandler);
+                pulsarAdmin, clientBuilder, fatalHandler, producerCache);
```

Comment:
I'm wondering if we should separate the producer cache exposed to user via ComtextImpl from the one used by PulsarSink at the framework level

Reply:
Good point. There's a single cache so that the size could be limited, but the cache keys are intended to be unique by passing the additional cache key as the cache key. I'll check if that's always the case and is sufficient.

Reply:
The caches are now properly separated. It's a single cache (so that we can limit the total size), but the cache key contains a cache area field to handle the separation.